### PR TITLE
genie: new, 1.28

### PR DESCRIPTION
**Pull request by neruthes**
Topic Description
-----------------

New packages:

- genie

Package(s) Affected
-------------------

- genie

Security Update?
----------------

No

Architectural Progress
----------------------

- [x] AMD64 `amd64`   
- [ ] AArch64 `arm64`
    

----

After the pull request is merged, all package(s) affected must be rebuilt against the `stable` Git tree and environment (only `stable` repository should be enabled in `sources.list`). This section marks the progress above.

Please, make sure the list of architectures below matches the ones above.

Post-Merge Architectural Progress
---------------------------------

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

Post-Merge Secondary Architectural Progress
-------------------------------------------

### CODE REVIEW

File: extra-utils/genie/autobuild/build [hunk 1]
```diff
@@ -0,0 +1,4 @@
+cd "$SRCDIR"/genie
```
Comment (MingcongBai) [inline extra-utils/genie/autobuild/build]: Move this to `spec` as `SUBDIR=genie/genie`.

File: extra-utils/genie/autobuild/build [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+
+make build
```
Comment (MingcongBai) [inline extra-utils/genie/autobuild/build:2]: Add some `abinfo` here please.

File: extra-utils/genie/autobuild/build [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+make install DESTDIR=$PKGDIR
```
Comment (MingcongBai) [inline extra-utils/genie/autobuild/build]: `$PKGDIR` should be quoted.

File: extra-utils/genie/autobuild/defines [hunk 4]
```diff
@@ -0,0 +1,4 @@
+PKGNAME=genie
+PKGSEC=utils
+PKGDEP="daemonize dbus dotnet-runtime-3.1 dotnet-sdk-3.1 systemd"
+PKGDES="A quick way into a systemd \"bottle\" for WSL"
```
Comment (MingcongBai) [inline extra-utils/genie/autobuild/defines]: This `PKGDES` does not make linguistic sense.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-utils/genie/autobuild/defines (hunk 4 above)
Comment (neruthes) [reply extra-utils/genie/autobuild/defines]: Updated

File: extra-utils/genie/autobuild/build (hunk 3 above)
Comment (neruthes) [reply extra-utils/genie/autobuild/build]: Fixed

File: extra-utils/genie/autobuild/build (hunk 2 above)
Comment (neruthes) [reply extra-utils/genie/autobuild/build:2]: Done

File: extra-utils/genie/autobuild/build (hunk 1 above)
Comment (neruthes) [reply extra-utils/genie/autobuild/build]: Done

File: extra-utils/genie/autobuild/defines [hunk 5]
```diff
@@ -0,0 +1,4 @@
+PKGNAME=genie
+PKGSEC=utils
+PKGDEP="daemonize dbus dotnet-runtime-3.1 dotnet-sdk-3.1 systemd"
+PKGDES="Fake systemd for WSL 2"
```
Comment (MingcongBai) [inline extra-utils/genie/autobuild/defines]: "Fake systemd for use with WSL 2" might work better.

File: extra-utils/genie/autobuild/build (hunk 2 above)
Comment (MingcongBai) [reply extra-utils/genie/autobuild/build:2]: Please separate `abinfo` for different build steps.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-utils/genie/autobuild/build (hunk 2 above)
Comment (neruthes) [reply extra-utils/genie/autobuild/build:2]: Done

File: extra-utils/genie/autobuild/defines (hunk 5 above)
Comment (neruthes) [reply extra-utils/genie/autobuild/defines]: Copied the explanation from upstream

**Review by MingcongBai — APPROVED** [review]

**Comment by Fearyncess** [comment]
genie works well. but it doesnt use `genie` command. it must use `dotnet /usr/libexec/genie/main/genie.dll` insteading.
image
image
image

**Review by MingcongBai — CHANGES_REQUESTED** [review]
Great. However, @liushuyu noted that the `arm64` package actually contains `amd64` binary. Further changes needed.

**Review by Fearyncess — CHANGES_REQUESTED** [review]
it lacks `genie` command directly to use. should it have a wrapper script?

**Comment by MingcongBai** [comment]
@neruthes Probably.

**Comment by liushuyu** [comment]
You may want to change the file https://github.com/arkane-systems/genie/blob/e88b1fbc8375ea4f4aec6638f7f032f5cdf46e86/genie/genie/Makefile#L10 in the repository according to https://docs.microsoft.com/en-us/dotnet/core/rid-catalog#linux-rids.

**Comment by neruthes** [comment]
Note:

The most significant usage is explained by <https://www.most-useful.com/ubuntu-20-04-desktop-gui-on-wsl-2-on-surface-pro-4.html#install_systemdgenie>.

Let me know if these is any other scenario beyond the scope of this article.

**Comment by OriginCode** [comment]
LGTM on amd64, but it lacks a wrapper script.
image

**Comment by liushuyu** [comment]
There is a test for aot-enabled compilation, please test this version:

https://gofile.io/d/ysxfyp

And if it works well, we can use this version and drop the `dotnet-runtime` dependency.

**Comment by Fearyncess** [comment]
i've tested this version and it works well on my WSL-2 instances. just use `genie -i`.
________________________________
发件人: liushuyu <notifications@github.com>
发送时间: 2020年11月27日 13:24
收件人: AOSC-Dev/aosc-os-abbs <aosc-os-abbs@noreply.github.com>
抄送: fearyncess <fsf@live.com>; Comment <comment@noreply.github.com>
主题: Re: [AOSC-Dev/aosc-os-abbs] genie: new, 1.28 (#2514)


There is a test for aot-enabled compilation, please test this version:

https://gofile.io/d/ysxfyp

And if it works well, we can use this version and drop the dotnet-runtime dependency.

―

**Comment by neruthes** [comment]
> There is a test for aot-enabled compilation, please test this version:
> 
> https://gofile.io/d/ysxfyp
> 
> And if it works well, we can use this version and drop the `dotnet-runtime` dependency.

I will try this tomorrow.

**Comment by neruthes** [comment]
Tested both the original version and the AOT version. Neither works on my WSL2. Very strange. Will investigate later.

**Comment by MingcongBai** [comment]
@OriginCode Sorry for a late bump, but what is that wrapper script you were referring to?

**Comment by OriginCode** [comment]
> @OriginCode Sorry for a late bump, but what is that wrapper script you were referring to?

https://github.com/AOSC-Dev/aosc-os-abbs/pull/2514#pullrequestreview-530776155

**Comment by liushuyu** [comment]
As of writing, `genie` [version 1.42 is already released](https://github.com/arkane-systems/genie/releases/tag/v1.42).

Marking as `stalled`.

**Comment by neruthes** [comment]
Should we be opening another pull request and abandon the current one?

**Comment by MingcongBai** [comment]
@neruthes Sure.

**Comment by liushuyu** [comment]
I am going to close this Pull Request as the original author said they will open a new one.

**Comment by MingcongBai** [comment]
@liushuyu Thank you senpai UwU

**Pull request closed without merging**